### PR TITLE
Add missing defaults to poisson_nll_loss

### PR DESCRIPTION
**Pull request by adampauls**
The defaults are present in the C++ and Python APIs, but not in native_functions.yaml, which means generated third-party bindings don't get them for free. Simply an oversight as far as I can tell. https://discuss.pytorch.org/t/torch-nn-functional-api-and-declarations-yaml/138374/2. 

cc @ezyang @gchanan

### CODE REVIEW

**Comment by albanD** [comment]
cc @mruberry Is there any  fc/bc concern with this?

**Comment by jbschlosser** [comment]
> cc @mruberry Is there any fc/bc concern with this?

I imagine so, if any usages of `torch.poisson_nll_loss()` exist out there

**Comment by adampauls** [comment]
Sorry, I wasn't careful there. I thought it was just an oversight that it wasn't declared to live in `nn` like the other losses as well. Happy to leave a method for backwards compatibility. Any other issues?

**Comment by albanD** [comment]
I don't think there is a bc concern for usage: any existing code calls with all the args and so it should be good.
I am more concerned about serialized models. Adding new defaults should not be a problem but wanted to confirm.

**Comment by jbschlosser** [comment]
> I don't think there is a bc concern for usage: any existing code calls with all the args and so it should be good.

What about the fact that `torch.poisson_nll_loss()` becomes `torch._C._nn.poisson_nll_loss()`? If the former is used anywhere, it will break.

Granted, users should ideally be using the `F.poisson_nll_loss()` form, which won't break, but we don't enforce that.

Update: I checked throughout GitHub and saw maybe 1 or 2 places at most where the name change would matter. Imo it's a fairly safe change to make, but it should still be flagged as bc-breaking to call out the change within the release notes. @albanD wdyt?

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -3445,7 +3445,8 @@
 - func: pinverse(Tensor self, float rcond=1e-15) -> Tensor
   variants: function, method
 
-- func: poisson_nll_loss(Tensor input, Tensor target, bool log_input, bool full, float eps, int reduction) -> Tensor
+- func: poisson_nll_loss(Tensor input, Tensor target, bool log_input=False, bool full=False, float eps=1e-08, int reduction=Mean) -> Tensor
```
Comment (mruberry) [inline aten/src/ATen/native/native_functions.yaml:3448]: There's already a `poisson_nll_loss` bound to the `torch.nn.functional` namespace with the correct defaults:

https://github.com/pytorch/pytorch/blob/7cdfd86a727ce01e41f2af81931064fb664513dd/torch/nn/functional.py#L2651

And here's the C++ API entry for completeness:

https://github.com/pytorch/pytorch/blob/828a9dcc048610bb5b45884634566093920b34cc/torch/csrc/api/include/torch/nn/functional/loss.h#L702

So I'm not sure what this PR is actually trying to accomplish for the user?

Comment (mruberry) [reply aten/src/ATen/native/native_functions.yaml:3448]: Note that `torch.poisson_nll_loss` is an undocumented function that, if anything, should be hidden from the user interface.

**Comment by albanD** [comment]
I agree the BC is good to remove an undocumented function.

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (adampauls) [reply aten/src/ATen/native/native_functions.yaml:3448]: I added a note to the description. The reason I added the defaults to the native functions declarations is so generated bindings get to make use of them. I agree that adding back `torch.poisson_nll_loss` manually to avoid breaking backwards compatibility is a good move and will do that soon.
